kernel/src/engine/arrow_utils.rs

```diff
@@ -69,6 +70,7 @@ pub(crate) fn make_arrow_error(s: impl Into<String>) -> Error {
 /// Applies post-processing to data read from parquet files. This includes `reorder_struct_array` to
 /// ensure schema compatibility, as well as `fix_nested_null_masks` to ensure that leaf columns have
 /// accurate null masks that row visitors rely on for correctness.
+#[internal_api]
```

Comment:
So, here be dragons (kinda).

The semantics of `request_ordering` are very specific, and it's easy to mess up and get panics and/or unexpected results. Really this is intended to be used in a flow of: `get_indices` -> [read parquet] -> `fixup_parquet_read`.

I'm  not opposed to exposing it necessarily, but it's a sharp edge on its own.

Reply:
As it turn out we don't even need to make this public, the thing I actually use is `fix_nested_null_masks`. For the parquet read we hook into the datafusion machinery. This did require hoisting some code from the `apply_schema_to` family of functions, but circumvents use of the parquet projection code here.